### PR TITLE
fix(tabs): fix nested interactive controls

### LINKED ISSUE #1585: [bug] `<rh-tabs>` per axeDevTools: Interactive controls must not be nested
@rmscampos is getting the following issue on the `<rh-tabs>` [demo](https://ux.redhat.com/elements/tabs/demo/):

```Interactive controls must not be nested```

**Screenshot:**
image

### PR DESCRIPTION
**Pull request by zeroedin**
## What I did

1. Corrects nested controls for `<rh-tab>`.  No longer embeds a `<button>` element.  This same fix was made upstream prior in `pf-tabs`.

Closes #1585

## Testing Instructions

1. View [deploy preview docs demo](https://deploy-preview-1603--red-hat-design-system.netlify.app/elements/tabs/demo/)

## Notes to Reviewers

Check automated accessibility tests no longer fail on nested interactive controls.

### CODE REVIEW

File: elements/rh-tabs/rh-tab.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export class TabExpandEvent extends Event {
  * @slot icon - Can contain an `<svg>` or `<pf-icon>`
  * @slot - Tab title text
  *
- * @csspart button - `<button>` element
+ * @csspart button - `<div>` element
```
Comment (bennypowers) [inline elements/rh-tabs/rh-tab.ts:37]:
```suggestion
 * @csspart button - element that contains the interactive part of a tab 
```

File: elements/rh-tabs/rh-tab.ts [hunk 2]
```diff
@@ -92,13 +89,13 @@ export class RhTab extends LitElement {
     const last = lastTab === this;
     return html`
       <div id="container" class="${classMap({ active, box, vertical, first, last, [on]: !!on })}">
-        <button part="button" ?disabled="${this.disabled}">
+        <div id="button" part="button" ?disabled="${this.disabled}">
```
Comment (bennypowers) [inline elements/rh-tabs/rh-tab.ts:92]: I wonder if we could combine this with the container element?

**Review by bennypowers — CHANGES_REQUESTED** [review]
One small docs change

File: elements/rh-tabs/rh-tab.ts (hunk 2 above)
Comment (zeroedin) [reply elements/rh-tabs/rh-tab.ts]: when you get the opportunity PTAL again!

File: elements/rh-tabs/rh-tab.css [hunk 3]
```diff
@@ -104,64 +122,32 @@ button:after {
 }
 
 /* Context switch on attributes */
-:host([active]) #container,
-#container.active {
+#button.active {
```
Comment (adamjohnson) [inline elements/rh-tabs/rh-tab.css:125]: The [APG recommends](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-automatic/) using [`aria-selected="true"` and `aria-selected="false"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-selected) to communicate to AT which tab is active. That said, that change might be out of scope for this PR.

**Review by adamjohnson — APPROVED** [review]
Logic Gives Tabs Meaning 👏

File: elements/rh-tabs/rh-tab.css (hunk 3 above)
Comment (zeroedin) [reply elements/rh-tabs/rh-tab.css:125]: This definitely was a regression, good catch!  Resolved in https://github.com/RedHat-UX/red-hat-design-system/pull/1603/commits/87e038b5210301ad01012f0a568350d315909831

Comment (bennypowers) [reply elements/rh-tabs/rh-tab.css:125]: Can you add an ax tree test for that using a11ySnapshot?

**Review by bennypowers — APPROVED** [review]
last good town meeting

**Pull request merged**